V 08 v 09 f 10

vladn255 opened the pull request:
На данный момент форма авторизации принимает любой валидный email и произвольную строку в качестве пароля.

vladn255 committed:
V_08_V-09_F-10

- migrated from sass to styled-components;
- made a page for orders display (a lot has happened here);

vladn255 committed:
V-08_V-09_F-10

- fixed select resetiing after re-render;
- templating conten-table;

vladn255 committed:
3. V-08_V-09_F-10

- added two more tables

vladn255 committed:
Update App.tsx

- forgot to return PrivateRoute

vladn255 committed:
V_08-V_09-F_10

- little fix

AlexsandrLitkevich commented on src/components/abstract-table/types.ts:
```diff
@@ -0,0 +1,37 @@
+    dataList: any[],
+    dataCount: number,
+    selectData: TSelectList,
+    selectMap: any,
```
почему ты используешь any ?

vladn255 replied:
В случаях, где код, в целом, рабочий, но нужные типы определить сходу не удаётся, пишу any. Понимаю, что это - малодушие и читерство с моей стороны, но иначе в сроки не укладываюсь)

AlexsandrLitkevich commented on src/components/cars-table/cars-table.tsx:
```diff
@@ -0,0 +1,96 @@
+
+    const selectMap = getMappedSelects(categorySelects)
+
+    const initСars = async () => {
```
асинхронные операции лучше делать вне компонента, используй redux thunk или  redux saga

AlexsandrLitkevich replied:
так как ты используешь redux toolkit то thunk там в коробке

AlexsandrLitkevich commented on src/components/category-table/types.ts:
```diff
@@ -0,0 +1,16 @@
+
+type TFilterData = IFilterDataItem[] | []
+
+interface ISelectOption {
```
тут можно и type использовать , почитай разницу типа и интерфейса. Еще с TS есть такие (Utility Types) и тут подошел бы Record ну это для кругозора

vladn255 committed:
Update package.json

AlexsandrLitkevich commented on src/components/logo-title/logo-title.tsx:
```diff
@@ -0,0 +1,36 @@
+
+import { useBoxStyles, useTextStyles, useIconStyles } from './styles'
+
+interface ILogoTitle {
```
на самом деле от I перед интерфейсом уже отказались мол избыточна но в код стал еще можно встретить

AlexsandrLitkevich replied:
для кругозора не обязательно исправлять

vladn255 committed:
Update const.ts

AlexsandrLitkevich commented on src/components/model-photo/model-photo.tsx:
```diff
@@ -0,0 +1,42 @@
+const FILE_NAME_SLICE = '/files';
+const URL_PART = 'https://api-factory.simbirsoft1.com';
+
+const checkIsValid = (src: string | undefined) => {
```
можно написать const checkIsValid = (src?: string) где ? делает параметр необязательным а смысл тот же только тебе не нужно  | undefined писать руками

AlexsandrLitkevich commented on src/components/model-photo/model-photo.tsx:
```diff
@@ -0,0 +1,42 @@
+const checkIsValid = (src: string | undefined) => {
+    let isValid = true;
+
+    if (typeof src !== 'string') {
```
а тут src можно просто просто проверить на true false так как TS пустить либо строку либо  undefined

AlexsandrLitkevich commented on src/components/model-photo/model-photo.tsx:
```diff
@@ -0,0 +1,42 @@
+    return isValid
+}
+
+const checkIsValidSrc = (src: string | undefined) => {
```
читай выше

AlexsandrLitkevich commented on src/components/order-table/order-table.tsx:
```diff
@@ -0,0 +1,94 @@
+        setIsLoading(true)
+
+        await Promise.all([
+            (async () => await dispatch(getOrders({ page: page, limit: LIMIT, filters: filterData })))(),
```
вынести из компонента

AlexsandrLitkevich commented on src/components/select-form/types.ts:
```diff
@@ -0,0 +1,25 @@
+interface IFormDataItem {
+    name: string,
```
типы так же можно переиспользовать

vladn255 replied:
Принято. Просто здесь простой интерфейс, так что, чтобы не плодить взаимные ссылки, прописал здесь.

AlexsandrLitkevich commented on src/components/select-input/types.ts:
```diff
@@ -0,0 +1,15 @@
+
+interface ISelectInput {
+    name: string,
+    value: string | undefined,
```
value?: string

vladn255 committed:
5. V_08-V_09-F_10

Bugfixes:
- removed async operations from table-components to slices,
- slight refactoring

vladn255 committed:
Merge branch 'Review' into V_08-V_09-F_10
